test-runner/src/main/scala/com/typesafe/spark/test/mesos/MesosIntTestHelper.scala

```diff
@@ -25,21 +25,21 @@ trait MesosIntTestHelper { self: FunSuite =>
    * Creates a SparkContext based on the given properties and reports test result to runner
    * @param name name of the job and mesos framework. The SPARK_FRAMEWORK_PREFIX-$name is used as the Spark
    *             application name.
-   * @param ps  key-value pairs of Spark configuration
+   * @param properties  key-value pairs of Spark configuration
    * @param t function that contains the testcase
    * @return ()
    */
-  def runSparkTest(name: String, ps: (String, String)*)(t: (SparkContext) => Unit) {
+  def runSparkTest(name: String, properties: () => Seq[(String, String)], tags: Seq[Tag] = Nil)(t: (SparkContext) => Unit) {
```

Comment:
This is a hack, but I made `properties` a function, because it's dependent on `cfg.role`, and `cfg` is now a function that only returns the config during test run, not when the suite is contstructed.  This is because config is now passed in via a `ConfigMap`.   See https://github.com/typesafehub/mesos-spark-integration-tests/pull/57/files#diff-c73a03c553794ebfb1476e14d64cad4cR28